service/controller/v15/cloudformation/cloud_formation.go

```diff
@@ -58,19 +58,55 @@ func (c *CloudFormation) DescribeOutputsAndStatus(stackName string) ([]*cloudfor
 	// be called during creation, while the outputs are still not accessible.
 	{
 		errorStatuses := []string{
+			cloudformation.StackStatusCreateInProgress,
 			cloudformation.StackStatusRollbackInProgress,
 			cloudformation.StackStatusRollbackComplete,
-			cloudformation.StackStatusCreateInProgress,
+			cloudformation.StackStatusUpdateInProgress,
 		}
 
+		// TODO to discuss in the PR:
+		//
+		//
+		// Complete states:
+		//
+		//	StackStatusDeleteComplete = "DELETE_COMPLETE"
+		//	StackStatusUpdateComplete = "UPDATE_COMPLETE"
+		//	StackStatusCreateComplete = "CREATE_COMPLETE"
+		//
+		//
+		// Rollback complete states:
+		//
+		//	StackStatusRollbackComplete = "ROLLBACK_COMPLETE"
+		//	StackStatusUpdateRollbackComplete = "UPDATE_ROLLBACK_COMPLETE"
+		//
+		//
+		// Failed states:
+		//
+		//	StackStatusCreateFailed = "CREATE_FAILED"
+		//	StackStatusRollbackFailed = "ROLLBACK_FAILED"
+		//	StackStatusDeleteFailed = "DELETE_FAILED"
+		//	StackStatusUpdateRollbackFailed = "UPDATE_ROLLBACK_FAILED"
+		//
+		//
+		// In progress states:
+		//
+		//	StackStatusCreateInProgress = "CREATE_IN_PROGRESS"
+		//	StackStatusDeleteInProgress = "DELETE_IN_PROGRESS"
+		//	StackStatusReviewInProgress = "REVIEW_IN_PROGRESS"
+		//	StackStatusRollbackInProgress = "ROLLBACK_IN_PROGRESS"
+		//	StackStatusUpdateCompleteCleanupInProgress = "UPDATE_COMPLETE_CLEANUP_IN_PROGRESS"
+		//	StackStatusUpdateInProgress = "UPDATE_IN_PROGRESS"
+		//	StackStatusUpdateRollbackCompleteCleanupInProgress = "UPDATE_ROLLBACK_COMPLETE_CLEANUP_IN_PROGRESS"
+		//	StackStatusUpdateRollbackInProgress = "UPDATE_ROLLBACK_IN_PROGRESS"
```

Comment:
Selection of `errorStatuses` feels incomplete. My common sense tells me we should add all `*_IN_PROGRESS`, all `*_FAILED`, all `*_ROLLBACK_*` and `DELETE_COMPLETE` states here.

Which boils down to only consider `UPDATE_COMPLETE` and `CREATE_COMPLETE` as not being in transition.
I would consider adding also `ROLLBACK_COMPLETE` and `UPDATE_ROLLBACK_COMPLETE` as we maybe need to reconcile other things after rollbacks to make sure we don't have downtimes. But we have `StackStatusRollbackComplete` excluded so it makes me uncertain.

I'm thinking of refactoring it to:

```go
okStatuses := []string{
	cloudformation.StackStatusCreateComplete,
	cloudformation.StackStatusUpdateComplete,
	cloudformation.StackStatusUpdateRollbackComplete, // maybe, uncertain
	cloudformation.StackStatusRollbackComplete, // maybe, uncertain
}

for _, s := range okStatuses {
	if stackStatus == s {
		break
	}
	return nil, microerror.Maskf(stackInTransitionError, "stack state '%s'", stackStatus)
}
```

That will allow to spot statuses which are maybe ok in the error and add them if we find them ok. But I think this strict list should be just fine.

Feedback appreciated.

Reply:
As long as we can access the outputs it is fine. I can just not guarantee this now. Returning `stackInTransitionError` when the rollback is complete does not sound right though. Cannot recall why we did this. It would semantically be better to distinguish these states so we can handle them or just rename the error we want to treat.

Reply:
I think I'll try with:

```
okStatuses := []string{
	cloudformation.StackStatusCreateComplete,
	cloudformation.StackStatusUpdateComplete,
}

for _, s := range okStatuses {
	if stackStatus == s {
		break
	}
	return nil, microerror.Maskf(stackInTransitionError, "stack state '%s'", stackStatus)
}
```

I will also restore returning `stackStatus` so we can log it to debug and slowly add to `okStatuses` as needed. That way we have we will have strict list and a way to extend it.

Reply:
`outputsNotAccessibleError` sounds good actually. :D

Reply:
Also, it is still this. 

```
		errorStatuses := []string{
			cloudformation.StackStatusCreateInProgress,
			cloudformation.StackStatusRollbackInProgress,
			cloudformation.StackStatusRollbackComplete,
			cloudformation.StackStatusUpdateInProgress,
		}
```

Reply:
The statuses in the comment look correct to me. When I tested deletion during an update the update failed because the ebsvolume resource shuts down the master. The stack went to `ROLLBACK_IN_PROGRESS` and then to `UPDATE_ROLLBACK_FAILED `.

Since the finalizer was kept the delete was repeated when the stack was in `UPDATE_ROLLBACK_FAILED `. Cluster was deleted with no leftovers. So with support for these status I think it looks good.

Reply:
Thanks for the input. I went with code below. Also logging the status where appropriate.

```
okStatuses := []string{
	cloudformation.StackStatusCreateComplete,
	cloudformation.StackStatusUpdateComplete,
}

for _, s := range okStatuses {
	if stackStatus == s {
		break
	}
	return nil, microerror.Maskf(stackInTransitionError, "stack state '%s'", stackStatus)
}```

Reply:
Looks like we need handle also RollbackComplete. I see it was introduced here https://github.com/giantswarm/aws-operator/pull/808/files#diff-2199eb13dbff5dd017d6e089240bb606R60 and never changed. We probably were lucky enough to not have to deal with ROLLBACK_COMPLETE in creation. This can easily happen in deletion when some resources are deleted before stack completes. Then stack rollbacks and deadlock. We never delete it, because we wait for it to be UPDATE_COMPLETE or CREATE_COMPLETE.